broker/submissions/export_to_spreadsheet_service.py

```diff
@@ -17,3 +24,44 @@ def export(self, submission_uuid: str):
         flattened_json = self.downloader.convert_json(entity_list)
         workbook = self.downloader.create_workbook(flattened_json)
         return workbook
+
+    def export_and_save(self, submission_uuid: str, storage_dir: str):
+        self.logger.info(f'Exporting submission {submission_uuid}')
+        try:
+            submission = self.ingest_api.get_submission_by_uuid(submission_uuid)
+            submission_url = submission['_links']['self']['href']
+        except Exception as e:
+            self.logger.error(e)
+            raise Exception(f'An error occurred in retrieving the submission with uuid {submission_uuid}: {str(e)}')
+
+        create_date = datetime.now(timezone.utc)
+
+        self._patch(submission_url, create_date)
+
+        directory = f'{storage_dir}/{submission_uuid}'
+        os.makedirs(f'{directory}/downloads/', exist_ok=True)
+
+        timestamp = create_date.strftime("%Y%m%d-%H%M%S")
+        filename = f'{submission_uuid}_{timestamp}.xlsx'
+        filepath = f'{directory}/downloads/{filename}'
+
+        workbook = self.export(submission_uuid)
+        workbook.save(filepath)
+
+        finished_date = datetime.now(timezone.utc)
+        self._patch(submission_url, create_date, finished_date)
```

Comment:
why patch create_date again?

Reply:
1st call patch to submission envelope resource:
```
{
    "lastSubmissionDownloadJob": {
        "createdDate": "timestamp",
        "finishedDate": null
    }
}
```

2nd call:

```
{
    "lastSubmissionDownloadJob": {
        "createdDate": "timestamp",
        "finishedDate": "timestamp2"
    }
}
```

If I don't include the `createdDate` in the second patch, it will be removed from the lastSubmissionDownloadJob object
```
{
    "lastSubmissionDownloadJob": {
        "finishedDate": "timestamp2"
    }
}
```